lib/companies_web/plugs/authenticate.ex

```diff
@@ -0,0 +1,24 @@
+defmodule CompaniesWeb.Plugs.Authenicate do
+  @moduledoc """
+  A simple plug to prevent users not logged in from continuing
+  """
+
+  import Phoenix.Controller, only: [put_flash: 3, redirect: 2]
+  import Plug.Conn
+
+  alias CompaniesWeb.Router.Helpers, as: Routes
+
+  def init(opts), do: opts
+
+  def call(%{assigns: %{current_user: %{}}} = conn, _opts) do
```

Comment:
Checks that we have a `current_user` that is a map. That's a simple enough check I think.